producers/prometheus/prometheus_test.go

```diff
@@ -17,16 +17,88 @@
 package prometheus
 
 import (
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
 	"testing"
+	"time"
 
 	"github.com/dcos/dcos-metrics/producers"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestNew(t *testing.T) {
 	Convey("The New method should return a producer and a channel", t, func() {
-		pi, pc := New()
+		pi, pc := New(Config{})
 		So(pi, ShouldHaveSameTypeAs, &promProducer{})
 		So(pc, ShouldHaveSameTypeAs, make(chan producers.MetricsMessage))
 	})
 }
+
+func TestRun(t *testing.T) {
+	Convey("Should create a Prometheus server on the configured port", t, func() {
+		// Ensure the environment is clean
+		if err := os.Unsetenv("LISTEN_PID"); err != nil {
+			panic(err)
```

Comment:
Is panicking here appropriate in a test?  In other words, do we want to use a test framework method to ensure that err is nil?

Reply:
I've been using panic when something has gone awry in test setup, but the code under test didn't fail (yet). Maybe it's better just to use assertions instead though 🤔